### ISSUE TITLE
Checkboxes generated with template not functional?

### ISSUE THREAD

**Issue by nightgolfer** [DylanGiesberts/obsidian-table-checkboxes#10]
After much frustration with checkboxes in tables, I finally found your plugin, and thought all my prayers were answered.

I confirmed that the plugin worked when I type in a checkbox in a table manually.

I then proceeded to write a template that loops over a DataviewJS query to generate table rows for each found item, and adding checkboxes to two cells in each row.

I figured I'd need to add IDs to the checkboxes for the plugin to work, so I wrote a little function to generate random 10-character IDs.

I've now got things mostly where I want them, and to my horror I've discovered that my checkboxes don't seem to work; the state is not retained between document loads, or view type switching.

I tried hand-typing a new checkbox into some of the cells, and they work as intended.

Am I right in assuming that the plugin only works with checkboxes it's converted to HTML itself? Or is there some way to make "existing" HTML checkboxes work?

If not, could that be a thing? Let the plugin scan a given page for checkboxes with an ID assigned, and manage their states?

**Edit**: Another option could be a shortcut for something like "parse document checkboxes", which could look for either markdown- or html checkboxes and "log/index" them (or whatever it is that the plugin needs to do to work). 

Even better, maybe some sort of tag in the frontmatter to tell the plugin to parse the page? That way, it can be added directly in the template, and any pages genereated by said template would work right off the bat, without having to call the proposed shortcut.

**Comment by DylanGiesberts** [comment]
For maintaining state, the plugin expects a checkbox to pass the following regex check:
`<input type="checkbox" (un)?checked id="${checkboxId}">`

Taking an example ID of '123', your checkboxes should look like one of the following:
- `<input type="checkbox" unchecked id="123">`
- `<input type="checkbox" checked id="123">`

You can try around here https://regex101.com/r/tniHoy/1

Can you check if it works in this format, and let me know if not?

**Comment by nightgolfer** [comment]
Thank you! 

Found the problem — I've been editing some ePub files for a client recently, and somehow got into a bad habit of using XHTML-style closing slashes. Here's a sample of one of my check boxes:

'<input type="checkbox" unchecked id="7357728273" />'

I removed the trailing slash after noticing that your regex checker wasn't accepting my sample. 

Everything works perfectly now, and I'm super excited to start using my template with persistent checkbox states! 😍

Again, thank you so much!
